llvm/lib/ProfileData/PGOCtxProfWriter.cpp

```diff
@@ -55,12 +67,16 @@ PGOCtxProfileWriter::PGOCtxProfileWriter(
                     SmallVector<unsigned, 1>({Version}));
 }
 
-void PGOCtxProfileWriter::writeCounters(const ContextNode &Node) {
+void PGOCtxProfileWriter::writeCounters(ArrayRef<uint64_t> Counters) {
   Writer.EmitCode(bitc::UNABBREV_RECORD);
   Writer.EmitVBR(PGOCtxProfileRecords::Counters, VBREncodingBits);
-  Writer.EmitVBR(Node.counters_size(), VBREncodingBits);
-  for (uint32_t I = 0U; I < Node.counters_size(); ++I)
-    Writer.EmitVBR64(Node.counters()[I], VBREncodingBits);
+  Writer.EmitVBR(Counters.size(), VBREncodingBits);
+  for (uint32_t I = 0U; I < Counters.size(); ++I)
+    Writer.EmitVBR64(Counters[I], VBREncodingBits);
```

Comment:
If you are touching this line, you might as well switch to a range-based for loop.

```suggestion
  for (uint64_t C : Counters)
    Writer.EmitVBR64(C, VBREncodingBits);
```

Reply:
good point, done